workitem/simple_type.go

```diff
@@ -94,6 +95,15 @@ func (fieldType SimpleType) ConvertToModel(value interface{}) (interface{}, erro
 		default:
 			return nil, errors.Errorf("value %v should be %s, but is %s", value, "MarkupContent", valueType)
 		}
+	case KindCodebase:
+		if valueType.Kind() != reflect.Map {
+			return nil, errors.Errorf("value %v should be %s, but is %s", value, reflect.Map, valueType.Name())
+		}
+		validCodebase, err := codebase.NewCodebase(value.(map[string]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		return validCodebase.ToMap(), nil
```

Comment:
Right, but the question is; Why does KindMarkup end up in this method as rendering.MarkupContent while KindCodebase comes in as a map.. ? Something is different somewhere. Probably how the type is converted from REST to Domain.